bundles/remote_services/remote_service_admin_dfi/src/remote_service_admin_dfi.c

```diff
@@ -952,7 +966,80 @@ celix_status_t remoteServiceAdmin_removeImportedService(remote_service_admin_t *
     return status;
 }
 
-static celix_status_t remoteServiceAdmin_send(void *handle, endpoint_description_t *endpointDescription, char *request, celix_properties_t *metadata, char **reply, int* replyStatus) {
+static celix_status_t remoteServiceAdmin_useDynamicIpUrlsForEndpoint(remote_service_admin_t* rsa, endpoint_description_t* endpointDescription,
+celix_status_t (*useUrl)(void* handle, const char* url), void* handle) {
+    celix_status_t status = CELIX_ERROR_MAKE(CELIX_FACILITY_CURL,CURLE_COULDNT_CONNECT);
+    celixThreadMutex_lock(&rsa->importedServicesLock);
+    celix_imported_endpoint_url_ref_t* urlRef = celix_stringHashMap_get(rsa->importedEndpointUrls, endpointDescription->id);
+    if (urlRef != NULL) {
+        celix_ref_get(&urlRef->ref);
+        celixThreadMutex_unlock(&rsa->importedServicesLock);
+        status = useUrl(handle, urlRef->url);
+        celix_ref_put(&urlRef->ref, (void*)free);
+    } else {
+        celixThreadMutex_unlock(&rsa->importedServicesLock);
+    }
+    if (status != CELIX_ERROR_MAKE(CELIX_FACILITY_CURL,CURLE_COULDNT_CONNECT)) {
```

Comment:
Is live lock possible here? That is, we are stuck with a URL that curl keeps returning an error other than `CURLE_COULDNT_CONNECT`.